### PR TITLE
Add urlquery function docs

### LINKED ISSUE #1627: Document `urlquery` function
Consider the following example:

```html
{{ $formatString := "https://www.facebook.com/sharer.php?u={url}" }}
{{ $result := replace $formatString "{url}" .Permalink }}
```

The issue I've had is that I want to "percent-encode" `.Permalink`. People have had this issue before:

* https://discourse.gohugo.io/t/urlencode-in-hugo/24215
* https://discourse.gohugo.io/t/is-there-a-url-encoder-function-in-hugo/2239/8

I've poked around the Hugo code base and found an undocumented built-in Hugo function `urlquery`:

https://github.com/gohugoio/hugo/blob/3d5dbdcb1a11b059fc2f93ed6fadb9009bf72673/tpl/internal/go_templates/texttemplate/funcs.go#L51

https://github.com/gohugoio/hugo/blob/3d5dbdcb1a11b059fc2f93ed6fadb9009bf72673/tpl/internal/go_templates/texttemplate/funcs.go#L738-L742

It's a thin wrapper around `url.QueryEscape` and perfectly solves my issue:

```html
{{ urlquery .Permalink }}
```

Is the function here to stay and just misses documentation?

### PR DESCRIPTION
**Pull request by schnerring**
Fixes #1627

### CODE REVIEW

File: content/en/functions/urlquery.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: urlquery
+linktitle: urlquery
+description: Takes one or more string inputs and returns a single percent-encoded string which is safe to use inside a URL query.
```
Comment (jmooring) [inline content/en/functions/urlquery.md:4]:
```suggestion
description: Returns the escaped value of the textual representation of its arguments in a form suitable for embedding in a URL query.
```

File: content/en/functions/urlquery.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+keywords: [urls]
+signature: ["urlquery INPUT [INPUT]..."]
+hugoversion:
+deprecated: false
+workson: []
+relatedfuncs: []
+aliases: []
+---
+
+`urlquery` takes one or more string inputs and returns a single [percent-encoded](https://en.wikipedia.org/wiki/Percent-encoding) string which is safe to use inside a URL query.
```
Comment (jmooring) [inline content/en/functions/urlquery.md:21]:
```suggestion
```

File: content/en/functions/urlquery.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+The following examples create a social sharing link for Facebook.
```
Comment (jmooring) [inline content/en/functions/urlquery.md:23]:
```suggestion
This template code:
```

File: content/en/functions/urlquery.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+```go-html-template
+<a href="https://www.facebook.com/sharer.php?u={{ urlquery "https://gohugo.io/" }}">Share</a>
```
Comment (jmooring) [inline content/en/functions/urlquery.md:26]:
```suggestion
{{ $u := urlquery "https://" "example.com" | safeURL }}
<a href="https://example.org?url={{ $u }}">Link</a>
```

File: content/en/functions/urlquery.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
```
Comment (jmooring) [inline content/en/functions/urlquery.md:27]:
```suggestion
```

File: content/en/functions/urlquery.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+<a href="https://www.facebook.com/sharer.php?u={{ urlquery "https://" "go" "hugo" ".io/" }}">Share</a>
```
Comment (jmooring) [inline content/en/functions/urlquery.md:28]:
```suggestion
```

File: content/en/functions/urlquery.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+```
+
+Both of these examples render the following HTML:
```
Comment (jmooring) [inline content/en/functions/urlquery.md:31]:
```suggestion
Is rendered to:
```

File: content/en/functions/urlquery.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+```html
+<a href="https://www.facebook.com/sharer.php?u=https%3A%2F%2Fgohugo.io%2F">Share</a>
```
Comment (jmooring) [inline content/en/functions/urlquery.md:34]:
```suggestion
<a href="https://example.org?url=https%3A%2F%2Fexample.com">Link</a>
```

**Review by jmooring — CHANGES_REQUESTED** [review]

**Comment by schnerring** [comment]
Sorry, I kinda couldn't decipher the suggested changes you made. It would nice if GitHub supported a combined overview the suggested changes to make this easier.

**Pull request merged**